conans/client/cmd/export.py

```diff
@@ -32,13 +32,25 @@ def cmd_export(conanfile_path, name, version, user, channel, keep_source,
     conan_linter(conanfile_path, output)
     conanfile = _load_export_conanfile(conanfile_path, output, name, version)
     conan_ref = ConanFileReference(conanfile.name, conanfile.version, user, channel)
+    if conan_ref.revision:  # Could be improved modifying the recipe?
+        raise ConanException("Specify the revision in a recipe field '_revision_'")
+    elif conanfile._revision_:
```

Comment:
The `_revision_` name could be changed, of course. Just take into account we could collide with some recipe using `revision`.